### DISCUSSION TITLE
is it possible to connect to 2 SQLite file db and do a join with them

### DISCUSSION THREAD

**Discussion by Tonic8** [mithrandyr/SimplySql#143] [Q&A] [answered]
i have a scenario where i have 2 SQLite db coming from diffrent source but would like to do a join between the data of the 2 DB

following this thread there is a command to attahc 2 db on the same connection
[SQLite - How do you join tables from different databases?](https://stackoverflow.com/questions/6824717/sqlite-how-do-you-join-tables-from-different-databases)

`
attach 'database1.db' as db1;
attach 'database2.db' as db2;
You can see all connected databases with keyword

.databases
Then you should be able to do the following.

select
  *
from
  db1.SomeTable a
    inner join 
  db2.SomeTable b on b.SomeColumn = a.SomeColumn;
`

i'm thinking to bulk copy to a temp SQLIte db the 2 db file, but if the file start to be huge, it will add more time to the scripts.

:)
Thanks

**Comment by mithrandyr** [comment, marked as answer]
@Tonic8 
You can use the `ATTACH` and `DETACH` commands.  however, make sure you pass in the full path to the file.  you can use `PRAGMA database_list` to see all the databases attached.

```powershell
Open-SQLiteConnection #open an in memory database
Invoke-SqlUpdate "ATTACH 'c:\fullpathtofile\somefile.db' AS db1"

Invoke-SqlQuery "PRAGMA database_list" #list out the databases on the current connnection
```
